### ISSUE TITLE
Deprecate and remove `SummaryPrinter`

### ISSUE THREAD

**Issue by mpkorstanje** [cucumber/cucumber-jvm#2054]
Prior to using events the `SummaryPrinter` interface provided a hook that could be used to print a summary. Now it only serves as a marker interface to indicate the plugin has something to print after a test run is done. This marker interface is used to determine if the default summary plugin should be replaced with a custom plugin. Now that only the CLI adds a default summary and progress reporter plugins the benefits of this system are marginal. 

For example:

```
java io.cucumber.core.cli.Main --plugin com.example.CustomSummary --plugin com.example.CustomFormatter
```

Will ensure that neither the default `summary` printer nor the default default `progress` formatter are used by:

```java
        RuntimeOptions runtimeOptions = commandlineOptionsParser
                .parse(argv)
                .addDefaultGlueIfAbsent()
                .addDefaultFeaturePathIfAbsent()
                .addDefaultFormatterIfAbsent()
                .addDefaultSummaryPrinterIfAbsent()
                .build(systemOptions);
```

The internal logic of Cucumber could be simplified by never enabling the default `progress` plugin and always enabling the default `summary` plugin`. The summary plugin would then have to be explicitly turned off via `--no-summary`.

This always prints a summary:

```
java io.cucumber.core.cli.Main
```

This prints a summary and progress dots:

```
java io.cucumber.core.cli.Main --plugin progress
```

This prints only progress dots

```
java io.cucumber.core.cli.Main --no-summary --plugin progress
```

This prints nothing

```
java io.cucumber.core.cli.Main --no-summary
```

This also prints nothing:

```
java io.cucumber.core.cli.Main --no-summary --plugin json:out.json
```

TODO:
 -  [ ] v6.x.x Deprecate `SummaryPrinter` interface
 -  [ ] v6.x.x  Add --[no-]summary comand line option
 -  [ ] v7.0.0 Remove `SummaryPrinter` and clean up plugin related parts of `RuntimeOptions*`.

**Comment by michiboo** [comment]
Hi Could I have a go at this issue?

**Comment by mpkorstanje** [comment]
@michiboo not really. This isn't scheduled until `v7.0.0` and we currently don't have a reason to make the next major version. Until that time this feature can hang around.
